modules/oe_theme_helper/tests/src/Functional/Plugin/field_group/InPageNavigationTest.php

```diff
@@ -0,0 +1,258 @@
+    'node',
+    'extra_field_test',
+    'field_test',
+    'field_group',
+    'oe_theme_helper',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Enable and set OpenEuropa Theme as default.
+    \Drupal::service('theme_installer')->install(['oe_theme']);
+    \Drupal::configFactory()->getEditable('system.theme')->set('default', 'oe_theme')->save();
+
+    // Create content type.
+    $this->type = 'first_node_type';
+    $this->drupalCreateContentType(['name' => 'Test type', 'type' => $this->type]);
+
+    // Create view display instance.
+    $display = \Drupal::entityTypeManager()
+      ->getStorage('entity_view_display')
+      ->load('node.' . $this->type . '.default');
+
+    // Create a node.
+    $node_values = ['type' => $this->type];
+
+    // Create 4 test fields.
+    for ($i = 0; $i < 4; $i++) {
+      $field_name = "field_test_$i";
+      $this->createTestField($field_name);
+
+      // Assign a test value for the field.
+      $node_values[$field_name][0]['value'] = ($i + 1) * 100;
+
+      // Set the field visible on the display object.
+      $display_options = [
+        'label' => 'above',
+        'type' => 'field_test_default',
+        'weight' => $i,
+        'settings' => [
+          'test_formatter_setting' => $this->randomMachineName(),
+        ],
+      ];
+      $display->setComponent($field_name, $display_options);
+    }
+
+    $display->save();
+    $this->node = $this->drupalCreateNode($node_values);
+  }
+
+  /**
+   * Tests the in-page navigation field group formatters.
+   *
+   * Structure of the content:
+   * Field group "In-page navigation group"
+   * - Field group "In-page navigation item"
+   * -- Field 0
+   * -- Field 1
+   * - Field group "In-page navigation item"
+   * -- Single text extra field
+   * -- Multiple items extra field
+   * - Field group "In-page navigation item"
+   * -- Field group "Html elements"
+   * --- Test field 2
+   * --- Test field 3
+   * - Field group "In-page navigation item"
+   * -- Formatted extra field without content.
+   */
+  public function testOutput(): void {
+    // Create "Html element" field group.
+    $children_group_html_element = [
+      'field_test_2',
+      'field_test_3',
+    ];
+    $group_html_element = $this->createFieldGroup('html_element', 'html_element', $children_group_html_element, 10, ['show_label' => TRUE]);
+
+    // Create "In-page navigation items" field groups.
+    $children_group_inpage_nav_items = [
+      [
+        'field_test_0',
+        'field_test_1',
+      ],
+      [
+        'extra_field_single_text_test',
+        'extra_field_multiple_text_test',
+      ],
+      [
+        $group_html_element->group_name,
+      ],
+      // It is empty extra field, so field group shouldn't be rendered.
+      [
+        'extra_field_empty_formatted_test',
+      ],
+    ];
+    $groups_inpage_nav_item = [];
+    $children_group_inpage_nav = [];
+    for ($i = 0; $i < 4; $i++) {
```

Comment:
I'm not a huge fan of this, if we add an extra navigation group to the test in the future we will need to update this. Instead we can do:    foreach ($children_group_inpage_nav_items as $index => $children_group_inpage_nav_item)